mpmath/matrices/matrices.py

```diff
@@ -676,8 +676,11 @@ def __rsub__(self, other):
         return -self + other
 
     def __eq__(self, other):
-        return self.__rows == other.__rows and self.__cols == other.__cols \
-               and self.__data == other.__data
+        try:
+            return (self.__rows == other.__rows and self.__cols == other.__cols
+                    and self.__data == other.__data)
+        except AttributeError:
+            return NotImplemented
```

Comment:
I'd not seen this before, but I skimmed https://stackoverflow.com/questions/40780004/returning-notimplemented-from-eq

I also tested this, and confirm that the `A9 == None` test below.